packages/bee-q/src/components/panel/bq-panel.tsx

```diff
@@ -0,0 +1,147 @@
+  shadow: true,
+})
+export class BqPanel {
+  // Own Properties
+  // ====================
+
+  private panel: HTMLElement;
+  private floatingUI: FloatingUI;
+  private trigger: HTMLElement;
+
+  // Reference to host HTML element
+  // ===================================
+
+  @Element() el!: HTMLBqPanelElement;
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  // Public Property API
+  // ========================
+
+  /** Distance (px) between the panel and the trigger element. */
+  @Prop({ reflect: true }) distance?: number = 0;
+
+  /** Position of the panel */
+  @Prop({ reflect: true }) placement?: FloatingUIPlacement = 'bottom';
+
+  /** If true, panel is visible.
+   * You can toggle this attribute to show/hide the panel.
+   */
+  @Prop({ reflect: true }) open?: boolean = false;
+
+  /** If true, the scrollbar is visible.
+   * You can toggle this attribute to show/hide the scrollbar.
+   */
+  @Prop({ reflect: true }) scrollbar?: boolean = false;
+
+  // Prop lifecycle events
+  // =======================
+
+  @Watch('open')
+  handleOpenProp() {
+    this.bqPanelVisibility.emit(this.open);
+  }
+
+  @Watch('distance')
+  @Watch('placement')
+  onPropChange() {
+    this.floatingUI.init({
+      placement: this.placement,
+      distance: this.distance,
+      sameWidth: false,
+      strategy: 'fixed',
+      skidding: 0,
+    });
+  }
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  /** Handler to be called to check if the panel is open or closed.
+   * Will emit every time the state of the panel is changed.
+   */
+  @Event() bqPanelVisibility: EventEmitter<boolean>;
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  disconnectedCallback() {
+    this.floatingUI?.destroy();
+  }
+
+  // Listeners
+  // ==============
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  @Method()
+  async togglePanel() {
+    this.floatingUI?.update();
+  }
+
+  /**
+   * set trigger element and init FloatingUI
+   * @param trigger - the trigger element for the panel
+   */
+  @Method()
+  async setTriggerElement(trigger: HTMLElement) {
+    this.trigger = trigger;
+    this.initFloatingUI();
+  }
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private initFloatingUI = () => {
+    this.floatingUI = new FloatingUI(this.trigger, this.panel, {
+      placement: this.placement,
+      distance: this.distance,
+      sameWidth: false,
+      strategy: 'fixed',
+      skidding: 0,
+    });
+  };
```

Comment:
I will propose adding here only the necessary private methods to handle:
- options mapping
- open the panel
- closing the panel

Something like:

```suggestion
  private get options() {
    return {
      distance: this.distance,
      placement: this.placement,
      sameWidth: this.sameWidth,
      skidding: this.skidding,
      strategy: this.strategy,
    };
  }

  private showPanel() {
    this.floatingUI?.update();
  }

  private async hidePanel() {
    this.open = false;
  }
```